Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Select.hs

```diff
@@ -185,14 +188,49 @@ getAddressByGetPlaceName merchantId sessionToken latLong = do
           sessionToken = Just sessionToken,
           language = Nothing
         }
-  mkLocation pickupRes
+  add <- mkLocation pickupRes
+  if (DT.length <$> add.area) /= Just 0
+    then pure (add, False)
+    else getAddressByNearBySearch merchantId add latLong
```

Comment:
if you return Nothing in else part and Just add in the then part, then you can handle it in the caller function using the code snipped shared in previous comment. These `getAddressByNearBySearch` and `getAddressByNearBySearch` need not have tuple of (Address, Bool) as their return type.